### DISCUSSION TITLE
Can not install on Android. Upgrade to Gradle 8.0

### DISCUSSION THREAD

**Discussion by u2gilles** [abhi3691/react-native-audio-remove#12] [Q&A]
Hi.
When I "npx expo run:android" ,   I have the error below. I think it has to do with Gradle 8.0. Could you please upgrade your project to Gradle 8.0. so that we can use it in 2023? Thanks in advance.

> Task :react-native-audio-remove:processDebugManifest
package="com.reactnativeaudioremove" found in source AndroidManifest.xml: D:\_P\dev\react\reactnative\Misc\fs1\node_modules\react-native-audio-remove\android\src\main\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.

> Task :app:processDebugMainManifest FAILED
package="com.fs1" found in source AndroidManifest.xml: D:\_P\dev\react\reactnative\Misc\fs1\android\app\src\main\AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
D:\_P\dev\react\reactnative\Misc\fs1\android\app\src\debug\AndroidManifest.xml Error:
        uses-sdk:minSdkVersion 21 cannot be smaller than version 24 declared in library [com.github.SimformSolutionsPvtLtd:SSffmpegVideoOperation:1.0.8] C:\Users\Gilles\.gradle\caches\transforms-3\084ec8f375a096f065bdaccf5df0ade1\transformed\jetified-SSffmpegVideoOperation-1.0.8\AndroidManifest.xml as the library might be using 
APIs not available in 21
        Suggestion: use a compatible library with a minSdk of at most 21,
                or increase this project's minSdk version to at least 24,
                or use tools:overrideLibrary="com.simform.videooperations" to force usage (may lead to runtime failures)

See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:processDebugMainManifest'.
> Manifest merger failed : uses-sdk:minSdkVersion 21 cannot be smaller than version 24 declared in library [com.github.SimformSolutionsPvtLtd:SSffmpegVideoOperation:1.0.8] C:\Users\Gilles\.gradle\caches\transforms-3\084ec8f375a096f065bdaccf5df0ade1\transformed\jetified-SSffmpegVideoOperation-1.0.8\AndroidManifest.xml as the library might be using APIs not available in 21
        Suggestion: use a compatible library with a minSdk of at most 21,
                or increase this project's minSdk version to at least 24,
                or use tools:overrideLibrary="com.simform.videooperations" to force usage (may lead to runtime failures)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 4m 9s
81 actionable tasks: 9 executed, 2 from cache, 70 up-to-date
Error: D:\_P\dev\react\reactnative\Misc\fs1\android\gradlew.bat exited with non-zero code: 1
Error: D:\_P\dev\react\reactnative\Misc\fs1\android\gradlew.bat exited with non-zero code: 1
    at ChildProcess.completionListener (D:\_P\dev\react\reactnative\Misc\fs1\node_modules\@expo\spawn-async\build\spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:628:26)
    at ChildProcess.emit (node:events:513:28)
    at cp.emit (D:\_P\dev\react\reactnative\Misc\fs1\node_modules\cross-spawn\lib\enoent.js:34:29)
    at Object.assembleAsync (D:\_P\dev\react\reactnative\Misc\fs1\node_modules\@expo\cli\build\src\start\platforms\android\gradle.js:52:18)
    at runAndroidAsync (D:\_P\dev\react\reactnative\Misc\fs1\node_modules\@expo\cli\build\src\run\android\runAndroidAsync.js:31:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
PS D:\_P\dev\react\reactnative\Misc\fs1>

**Comment by abhi3691** [comment]
Ok i will update

**Comment by u2gilles** [comment]
And I will test it again.  Let me know please.

PS: I said the error may be related to your Gradle version but I am not sure at all about it. It just does not build in my case.
